src/app/pokemon.service.ts

```diff
@@ -15,12 +18,23 @@ const httpOptions = {
 })
 export class PokemonService {
 
-  private _baseUrl: string = 'https://pokeapi.co/api/v2'
-  private _spriteBaseUrl: string = 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other-sprites/official-artwork';
+  private _baseUrl: string = 'https://pokeapi.co/api/v2';
 
-  constructor( private http: HttpClient ) { }
+  constructor(private http: HttpClient) {
+  }
+
+  findAll(offset: number = 0, limit: number = 20): Observable<PokeApiResponse> {
+    return this.http
+      .get<PokeApiResponse>(`${this._baseUrl}/pokemon/?offset=${offset}&limit=${limit}`, httpOptions)
+      .pipe(
+        map(response => {
```

Comment:
map allows us to transform the data before it gets subscribed to inside of the async pipe. Great place to transform the raw data in to pokemon entry objects.

Reply:
refer to this document to learn about the `pipe` and `map` operators. I even had some trouble with this, RXJS recently changed the api so this page sadly may be the easiest way to learn it.

https://angular.io/guide/rx-library